internal/ingress/controller/controller.go

```diff
@@ -237,12 +237,22 @@ func (n *NGINXController) CheckIngress(ing *networking.Ingress) error {
 	cfg := n.store.GetBackendConfiguration()
 	cfg.Resolver = n.resolver
 
-	for key := range ing.ObjectMeta.GetAnnotations() {
+	arraybadWords := strings.Split(strings.TrimSpace(cfg.AnnotationValueWordBlocklist), ",")
+
+	for key, value := range ing.ObjectMeta.GetAnnotations() {
+
 		if parser.AnnotationsPrefix != parser.DefaultAnnotationsPrefix {
 			if strings.HasPrefix(key, fmt.Sprintf("%s/", parser.DefaultAnnotationsPrefix)) {
 				return fmt.Errorf("This deployment has a custom annotation prefix defined. Use '%s' instead of '%s'", parser.AnnotationsPrefix, parser.DefaultAnnotationsPrefix)
 			}
 		}
+		if strings.HasPrefix(key, fmt.Sprintf("%s/", parser.AnnotationsPrefix)) {
+			for _, forbiddenvalue := range arraybadWords {
+				if strings.Contains(value, forbiddenvalue) {
```

Comment:
My only thought is if we need to run `strings.TrimSpace` every element as well, I’m afraid `Contains` may fail otherwise.

Reply:
can you give me one example? not following here (rainy day, a bit sleepy... :) )

Reply:
Here’s what I mean:
https://play.golang.org/p/mqWmLjoz63N

Reply:
@theunrealgeek GOOD CATCH!!! Thanks! I will quickly fix this and also add some regression / e2e test before releasing!

Reply:
But now I see that I do trimming a bit upper:
```
arraybadWords := strings.Split(strings.TrimSpace(cfg.AnnotationValueWordBlocklist), ",")
```

Don't this solve the problem?

Reply:
@rikatz 
```
arraybadWords := strings.Split(strings.TrimSpace(cfg.AnnotationValueWordBlocklist), ",")
```
will only trim space at the beginning and end of the annotation value, but not in between elements that may exist around the commas which is what my example in the Go playground is pointing out. 

In the default value this won’t be a problem since you are coming the CSV with a `Join` on the list, but for user overridden ones it will be hard to enforce that they don’t put spaces between the commas to make things more readable.

Reply:
ok, I see :) I was in my head thinking that trimSpace works for all the array, not only beginning and ending and you are right :)

Weirdly, when adding this to unit_test it passes as well, will have to check it better.

Reply:
Anyway, PTAL here: https://github.com/kubernetes/ingress-nginx/pull/7921